src/UnigramWordPredictor.java

```diff
@@ -52,60 +54,91 @@ public void train(Scanner scanner) {
     List<String> trainingWords = tokenizer.tokenize(scanner);
 
     // TODO: Convert the trainingWords into neighborMap here
+    //Map<String, List<String>> neighborMap = new HashMap<>();
+    this.neighborMap = new HashMap<>();
```

Comment:
This works well! You can make it a bit simpler by omitting the `this`.
```
neighborMap = new HashMap<>();
```
It does the same thing: for any variables that are not otherwise declared in the local scope, Java assumes they are instance variables and essentially puts an invisible `this.` in front of them.